src/Service/Parser/FunctionParser/ExplicitFunctionDefinitionParser.php

```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service\Parser\FunctionParser;
+
+use Lmc\ApiFilter\Assertion;
+use MF\Collection\Immutable\Tuple;
+
+class ExplicitFunctionDefinitionParser extends AbstractFunctionParser
+{
+    /**
+     * @param string|array $rawValue Raw value from query parameters
+     */
+    protected function supportsParameters(array $queryParameters, string $rawColumn, $rawValue): bool
+    {
+        return !$this->isTuple($rawColumn) && $this->functions->isFunctionRegistered($rawColumn);
+    }
+
+    /**
+     * @param string|array $rawValue Raw value from query parameters
+     */
+    protected function parseParameters(array $queryParameters, string $rawColumn, $rawValue): iterable
+    {
+        if (!$this->functions->isFunctionRegistered($rawColumn)) {
+            return;
+        }
+
+        yield from $this->parseFunction($rawColumn);
+
+        $parameters = $this->functions->getParametersFor($rawColumn);
+        if (count($parameters) === 1) {
+            Assertion::false(
```

Comment:
Dtto, use assertion chain instead?

Also is seems `isTuple` could be custom assertion itself.

Reply:
As I said before, chain for custom assertion is not easy.
And this assertion is different from that before - here the value must NOT be a tuple or array - it means it must be a single string value - I've moved it to a private method with that name, but I wont abstract it more.